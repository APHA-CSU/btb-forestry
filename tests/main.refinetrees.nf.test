nextflow_process {

    name "Test Process refinetrees"
    script "main.nf"
    process "refinetrees"

    test("Should run tree refinment") {

        when {
            params {

            }
            process {
                """
               input[0] = [ 'test', 
                            file("${projectDir}/tests/data/MP.nwk"),
                            '52532',
                            'AF-16-01234-12',
                            'test',
                            file("${projectDir}/tests/data/snp-only.fas") 
                            ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
