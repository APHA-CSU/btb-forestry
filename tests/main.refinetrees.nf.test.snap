{
    "Should run tree refinment": {
        "content": [
            {
                "0": [
                    [
                        "test",
                        "test_22Aug25_MP-rooted.nwk:md5,746e9af3d26b59232ea35c2e9d837a48",
                        "test_22Aug25_phylo.json:md5,0914cace1bff9dedc002f063d7128ffe"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-22T15:34:02.449855461"
    }
}